src/shell.ts

```diff
@@ -0,0 +1,96 @@
+'use strict';
+
+import { ChildProcess } from 'child_process';
+import * as vscode from 'vscode';
+import * as shelljs from 'shelljs';
+
+export enum Platform {
+    Windows,
+    MacOS,
+    Linux,
+    Unsupported,  // shouldn't happen!
+}
+
+export interface ExecCallback extends shelljs.ExecCallback {}
+
+export interface Shell {
+    isWindows(): boolean;
+    isUnix(): boolean;
+    home(): string;
+    exec(cmd: string, stdin?: string): Promise<ShellResult | undefined>;
+    execCore(cmd: string, opts: any, callback?: (proc: ChildProcess) => void, stdin?: string): Promise<ShellResult>;
+}
+
+export const shell: Shell = {
+    isWindows : isWindows,
+    isUnix : isUnix,
+    home : home,
+    exec : exec,
+    execCore : execCore
+};
+
+const WINDOWS: string = 'win32';
+
+export interface ShellResult {
+    readonly code: number;
+    readonly stdout: string;
+    readonly stderr: string;
+}
+
+export type ShellHandler = (code: number, stdout: string, stderr: string) => void;
+
+function isWindows(): boolean {
+    return (process.platform === WINDOWS);// && !getUseWsl();
```

Comment:
what means this commented out code?

Reply:
Removed. Cleaned up the shell file to eliminate all the things we're not using.